sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -212,7 +212,7 @@ def _format_url(self, hostname):
             self.scheme,
             hostname,
             quote(container_name),
-            quote(self.blob_name, safe='~'),
+            quote(self.blob_name, safe="~()$=',"),
```

Comment:
Let's revert this change ^_^ 
I think the problem is string_to_sign changed after we added AsyncTransportWrapper.
see [construct string_to_sign](https://github.com/Azure/azure-sdk-for-python/blob/e2266ebfed086fec1ac50fdd5aaef9cbc522efe8/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L79)
After we added AsyncTransportWrapper we should check if the internal property _transport is AioHttpTransport.

Probably the code should be:
`            if isinstance(request.context.transport, AioHttpTransport) \
                    or hasattr(request.context.transport, '_transport') \
                    and isinstance(request.context.transport._transport, AioHttpTransport):
                uri_path = URL(uri_path)`

Reply:
That's right. https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/azure/storage/blob/_shared/authentication.py#L79

this line is causing the error after we added the AsyncTransportWrapper. That was exactly the fix i was about to push

Reply:
That's great!!